docs/content/basics/testing_theory.md

```diff
@@ -1,3 +1,57 @@
 # Testing theory
 
 
+This article will cover the most primitive things that will allow you to understand better the terms and general concepts regarding testing.
+<br/>It is also worth noting that there will be an accent on UI tests, because we are in CookBook for UI tests.
+
+# Where it all starts
+<i>Why do I need testing?</i> Users of your application/system/whatever should receive a quality product. 
+<br/>Thus, testing is about ensuring the quality of the product. Testing also includes creating a test plan, creating/conducting tests themselves and analyzing results of testing.
+
+Let's focus on the topic of creating tests. To begin with, let's see the testing pyramid.
+
+<image here>
+
+The width of each block is actually the ratio of the number of different types of tests to each other. 
+For example, a lot of Unit tests is usually considered correct in the pyramid, but much less UI tests. 
+<br/><br/><b>This is due to two key parameters - stability and the cost of supporting each type of testing.</b>
+<br/>Unit tests have the highest stability, they are the fastest and they have the lowest cost of support, however, using Unit tests you won't be able to check the login flow of your application (which for example contains 5 screens).
+<br/><br/>We will reveal each type of testing in more detail, for this we will go from the bottom up.
+
+!!! Disclaimer
+
+    <b>White-box testing</b> is a type of testing in which we have full access to the implementation and can interact with it. We know which output data will be with given input data.
+    <br/><b>Black-box testing</b> is a type of testing in which we don't have access to the implementation and cannot interact with it, however, we know which output data should be with given input data.
+    <br/><b>Gray-box testing</b> is a rare type of testing when you have partial access to the implementation (for example, not to all entities that being tested). At the same time, we know what output data will be with given input data.
```

Comment:
They are not `rare` in our mobile world to be honest, the most of the companies use them.

I'd propose to remove `rare`